### DISCUSSION TITLE
Supporting variable number of arguments

### DISCUSSION THREAD

**Discussion by ttytm** [webui-dev/webui#210] [Ideas]
Would it be feasible to implement support for a variable number of arguments? So that call would be updated from
```typescript
call(bindingName: string, payload?: Payload): Promise<Response | void> {
```
to
```typescript
call(bindingName: string, ...payload?: Payload): Promise<Response | void> {
```

`webui_get_string()` and equivalents could accept an index as argument `webui_get_string(0)`.

**Comment by hassandraga** [comment]
I guess yes, it's a little bit challenging and time consuming... but it's a nice feature to have.
Probably I can spend time implementing this feature and then we release the v2.4.0 with it.

**Reply by ttytm** [reply]
If it doesn't fit in this release it would also be fine in a next version I think 👍

**Comment by hassandraga** [comment]
I'm going to remove all JSON functionality from WebUI because it will be replaced by this new multi argument feature. This means the wrappers need to remove it as well. If you have another idea or I missed something, please let me know.

**Comment by hassandraga** [comment]
Also, `.data` and `.size` will be removed from `webui_event_t`. The only way to get data is by using `webui_get_xxx(y)`, and to get the size is trough `webui_get_size(y)`.

This is a "_big_" breaking change... I vote to implement this feature in `v2.4.1`. What do you think @ttytm?

**Reply by ttytm** [reply]
It's indeed a change that spans alot.

But I think since the 2.4.0 release already has some breaking changes and breaking changes usually are not part of `x.x.<patch>` versions it would fit into the 2.4.0 release.

If not in the upcoming release than I would vote for a 2.5.0.

**Reply by hassandraga** [reply]
You are right. I forgot about that. So, we will keep fixes, if any, under `2.4.x`. 
This new feature will be implemented in `2.5.0`.

**Comment by hassandraga** [comment]
Implemented (https://github.com/webui-dev/webui/commit/b09c0dca947a882a8f48e408b076a595612b710c).

```c
int arg_1 = webui_get_int_at(e, 0);
int arg_2 = webui_get_int_at(e, 1);
...
```

**Reply by ttytm** [reply]
Those are some brutal changes recently. But I like it as it's for the good and the version already had several breaks. So it fits well.

Going soft and following semver after 2.4.0 stable would be a great step then.

Thanks for the work @hassandraga :+1:

**Reply by hassandraga** [reply]
I'm going to update the C++. Then let's do some tests, and if everything goes well, then we release `v2.4.0`. After that, we can apply patches `v2.4.x` if we miss any bugs.

**Reply by ttytm** [reply]
Sounds great. By chance give me a 👍 or 👎 if the go wrapper should be updated to not use static lib files anymore but go generate to download the latest webui releases. I added an example over at the release downloader WIP PR.

**Reply by hassandraga** [reply]
We should remove all binaries from all wrappers. Even the online package managers should be updated by CI. I'm looking forward to seeing the final implementation in the Go wrapper 👍
